syntax/scan.go

```diff
@@ -196,6 +200,13 @@ func (p Position) String() string {
 	return fmt.Sprintf("%s:%d:%d", p.Filename(), p.Line, p.Col)
 }
 
+func (p Position) IsBefore(p2 Position) bool {
```

Comment:
This method needn't be exported: isBefore

Call p2 "q".

Reply:
Done